### PR TITLE
Remove explosion-protected-regions

### LINKED ISSUE #230: EntityExplodeEvent called twice
If a TNT explodes the EntityExplodeEvent is called twice first with a blockList containing only one AIR block and the second time with the actual blocklist.

```
    @EventHandler
    public void onEntityExplode(EntityExplodeEvent e) {
        System.out.println(e.blockList());
    }
```

Produces the following output

```
[01:56:01] [Server thread/INFO]: [CraftBlock{chunk=CraftChunk{x=-25z=8},x=-389,y=62,z=141,type=AIR,data=0}]
[01:56:01] [Server thread/INFO]: [CraftBlock{chunk=CraftChunk{x=-25z=8},x=-388,y=61,z=138,type=DIRT,data=0}, CraftBlock{chunk=CraftChunk{x=-25z=8},x=-390,y=65,z=143,type=DIRT,data=0}, ...}]
```
[nacho.txt](https://github.com/CobbleSword/NachoSpigot/files/7531148/nacho.txt)

### LINKED ISSUE #302: TNT explosions not dealing damage to blocks
### Observed Behavior

Blocks not broking after TNT explosions

### Expected Behavior

block should be broken

### Steps To Reproduce

run a nachospigot server, with a plugin that causes explosion and this bug happenes

### Plugin List

_No response_

### Server Version

[21:55:11 INFO]: This server is running NachoSpigot version git-NachoSpigot-625c213 (MC: 1.8.8) (Implementing API version 1.8.8-R0.2-SNAPSHOT)

### Other

_No response_

### Agreements

- [X] You were able to find this issue on the latest version of NachoSpigot.
- [X] You have confirmed that there aren’t any issues open regarding this bug.
- [X] You have confirmed that you are NOT using a fork of NachoSpigot in any way. **YOUR changes are not OUR faults.**

### PR DESCRIPTION
**Pull request by Elierrr**
# Description

Removes the configuration option "explosion-proteccted-regions"

Fixes #230
Fixes #302
Closes #303 

# Additional Comments

I'm not exactly sure what this is for however it doesn't seem to be needed.

# Checklist:

- [x] I have reviewed my code thoroughly.
- [x] I have tested my code.
- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by Lucaskyy** [comment]
I assume it's for things such as spawn protection? So it allows you to blow up chunks that are spawn protected.

File: NachoSpigot-Server/src/main/java/net/minecraft/server/Explosion.java [hunk 1]
```diff
@@ -65,28 +63,10 @@ public void a() {
...
-                List<org.bukkit.block.Block> list = new java.util.ArrayList<>(1);
-                int x = org.bukkit.util.NumberConversions.floor(posX);
-                int y = org.bukkit.util.NumberConversions.floor(posY);
-                int z = org.bukkit.util.NumberConversions.floor(posZ);
-                list.add(chunk.bukkitChunk.getBlock(x, y, z));
-
-                EntityExplodeEvent event = new EntityExplodeEvent(source.getBukkitEntity(), location, list, 0.3F);
-                world.getServer().getPluginManager().callEvent(event);
-                if (event.isCancelled() || event.blockList().isEmpty()) {
-                    protection = true;
```
Comment (Elierrr) [inline NachoSpigot-Server/src/main/java/net/minecraft/server/Explosion.java:81]: From the code it looks like it's checking if the plugin cancels it. @Lucaskyy

Comment (Lucaskyy) [reply NachoSpigot-Server/src/main/java/net/minecraft/server/Explosion.java:81]: Oh, I see. So that config basically "bypasses" the event cancellation?

Comment (Elierrr) [reply NachoSpigot-Server/src/main/java/net/minecraft/server/Explosion.java:81]: Seems like it.

**Comment by CyberFlameGO** [comment]
Y’know you don’t need to always force-push everything, it’s only useful when you’re squashing a few commits (3+) that are just edits to the same change, or removing private info.

**Review by CyberFlameGO — APPROVED** [review]
Looks good to me!

**Comment by Elierrr** [comment]
> Y’know you don’t need to always force-push everything, it’s only useful when you’re squashing a few commits (3+) that are just edits to the same change, or removing private info.

I prefer having everything in one commit instead of multiple, and also removing private info wouldn't work since you can still view older commits in GitHub.

**Comment by CyberFlameGO** [comment]
Fair enough, but in my opinion it’s best to have a feature per commit so if people need to revert something it doesn’t revert useful changes. I forgot that PRs track commits though; I was thinking of repos where you can remove private info via force-push

**Comment by sadcenter** [comment]
I had the same thought in my mind when I was making the #234 patch. I thought I just don't understand why is it here and should leave it.

**Comment by Lucaskyy** [comment]
If there are no further comments/reviews I'll merge this PR this afternoon.

**Comment by Lucaskyy** [comment]
I see there are conflicts, could you look into it @Elierrr ?

**Comment by Elierrr** [comment]
> I see there are conflicts, could you look into it @Elierrr ?

Fixed the conflicts.

**Comment by Lucaskyy** [comment]
Thought of that ^ when checking the code. I edited it from GitHub nor did I test it, but I think it shouldn't cause any problems.

**Comment by Lucaskyy** [comment]
Anyway, I'm going to merge this once the build succeeds.
Thank you @Elierrr for this PR! :)

**Comment by Lucaskyy** [comment]
Oops, gotta fix compile errors hehe.

**Pull request merged**